src/allmydata/test/test_eliotutil.py

```diff
@@ -281,3 +282,75 @@ def f():
                 ),
             ),
         )
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_gets_positional_arguments(self, logger):
+        """
+        Check that positional arguments are logged when using ``log_call_deferred``
+        """
+        @log_call_deferred(action_type=u"the-action")
+        def f(a):
+            return a ** 2
+        self.assertThat(
+            f(4), succeeded(Equals(16)))
+        msg = logger.messages[0]
+        assertContainsFields(self, msg, {"args": (4,)})
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_gets_keyword_arguments(self, logger):
+        """
+        Check that keyword arguments are logged when using ``log_call_deferred``
+        """
+        @log_call_deferred(action_type=u"the-action")
+        def f(base, exp):
+            return base ** exp
+        self.assertThat(f(exp=2,base=10), succeeded(Equals(100)))
+        msg = logger.messages[0]
+        assertContainsFields(self, msg, {"kwargs": {"base": 10, "exp": 2}})
+
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_gets_keyword_and_positional_arguments(self, logger):
+        """
+        Check that both keyword and positional arguments are logged when using ``log_call_deferred``
+        """
+        @log_call_deferred(action_type=u"the-action")
+        def f(base, exp, message):
+            return base ** exp
+        self.assertThat(f(10, 2, message="an exponential function"), succeeded(Equals(100)))
+        msg = logger.messages[0]
+        assertContainsFields(self, msg, {"args": (10, 2)})
+        assertContainsFields(self, msg, {"kwargs": {"message": "an exponential function"}})
+
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_keyword_args_dont_overlap_with_start_action(self, logger):
```

Comment:
I don't really understand what this is testing.

Reply:
Now that I look at it again, the function is poorly named.

Actually, it's meant to test keyword arguments passed to decorated functions do not overlap with any keyword arguments passed directly to `start_action` on line [323](https://github.com/tahoe-lafs/tahoe-lafs/blob/07ffde3bec2436e855edaae0b0a150599280ac92/src/allmydata/util/eliotutil.py#L323) in eliotutil.